src/main/java/seedu/address/model/iteration/Iteration.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model.iteration;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents an Iteration in the ArtBuddy.
+ * Guarantees: date and description are present and not null, field values are validated, immutable.
+ */
+public class Iteration {
+
+    // Identity fields
+    private final Date date;
+    private final IterationDescription description;
+    // TODO add artwork (image) attribute
+
+    // Optional fields
+    private final Feedback feedback;
+
+    /**
+     * Constructs an Iteration.
+     * Every field must be present and not null. Throws a NullPointerException if
+     * any of the fields are null.
+     *
+     * @param date The creation date of the iteration.
+     * @param description The description of the iteration.
+     */
+    public Iteration(Date date, IterationDescription description, Feedback feedback) {
+        requireAllNonNull(date, description, feedback);
+        this.date = date;
+        this.description = description;
+        this.feedback = feedback;
+    }
+
+    public Date getDate() {
+        return date;
+    }
+
+    public IterationDescription getDescription() {
+        return description;
+    }
+
+    public Feedback getFeedback() {
+        return feedback;
+    }
+
+    /**
+     * Returns true if both iterations have the same image.
+     * This defines a weaker notion of equality between two iterations.
+     */
+    public boolean isSameIteration(Iteration otherIteration) {
+        if (otherIteration == this) {
+            return true;
+        }
+
+        // return otherIteration != null
+        //         && otherIteration.getArt().equals(getArt());
+        // TODO implement isSameIteration once the artwork/ image attribute has been added
+        return otherIteration != null
+                && otherIteration.getDescription().equals(getDescription());
+    }
```

Comment:
So subsequently will description still be used fro comparison?

Reply:
For now we can use the description as comparison. I think using file names will still have problems because I expect many users to just overwrite their image file when saving new one.

Reply:
Here's a possibility for checking if 2 images are the same: https://github.com/KilianB/JImageHash